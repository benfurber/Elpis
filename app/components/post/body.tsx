import React from "react";
import { StyleSheet, Text, View } from "react-native";
import FlexImage from "react-native-flex-image";

import { Post } from "interfaces";
import { colours, layout } from "styles";

const formatDate = (date: Date) => {
  const formatOptions = {
    day: "2-digit",
    month: "2-digit",
    year: "numeric"
  };
  return date.toLocaleString("en-GB", formatOptions);
};

interface Props {
  date: Post["date"];
  description: Post["description"];
  imagePath: Post["imagePath"];
}

const Body = (props: Props) => {
  const renderBodyText = () => {
    if (props.description) {
      return <Text style={styles.text}>{props.description}</Text>;
    }
  };

  return (
    <View style={styles.body}>
      <FlexImage source={props.imagePath} style={styles.image} />
      {renderBodyText()}
      <Text style={[styles.date, styles.text]}>{formatDate(props.date)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: colours.whiteTransparent,
    borderTopRightRadius: layout.borderRadius,
    padding: layout.padding,
    width: "100%"
  },
  date: {
    fontStyle: "italic"
  },
  text: {
    marginVertical: 10
  },
  image: {
    borderRadius: layout.borderRadius,
    overflow: "hidden"
  }
});

export { Body };
