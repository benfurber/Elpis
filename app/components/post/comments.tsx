import React from "react";
import { Image, StyleSheet, Text, View, ScrollView } from "react-native";

import { Post } from "interfaces";
import { colours, layout, typography } from "styles";
import { firstSentence } from "utils";

interface Props {
  description: Post["description"];
}

const Comments = (props: Props) => {
  const renderDescription = () => {
    if (props.description) {
      return (
        <View style={styles.titleContainer}>
          <Text style={styles.title}>{firstSentence(props.description)}</Text>
        </View>
      );
    }
  };

  return (
    <View style={styles.container}>
      <ScrollView>
        {renderDescription()}
        <View style={styles.commentsHeadingContainer}>
          <Text style={styles.commentsHeading}>x Comments - x Topics</Text>
        </View>
        <View style={styles.commentContainer}>
          <View style={styles.avatarContainer}>
            <View style={styles.badge}>
              <Text style={styles.badgeText}>4</Text>
            </View>
            <View>
              <Image
                source={require("assets/images/profile-pic-may.jpg")}
                style={styles.image}
              />
            </View>
          </View>
          <View style={styles.commentBodyContainer}>
            <Text style={styles.commentTitle}>
              Meu pai fez o que ela mandou…
            </Text>
            <Text style={styles.commentDate}>Há 7 minutos atrás</Text>
            <Text>
              Dá certo sim, o meu pai, por exemplo, fugiu quando eu tinha 5 anos
              e eu não faço ideia da onde ele esteja. Nunca mais voltou, pena
              que a violência só piorou.
            </Text>
          </View>
        </View>
        <View style={styles.commentContainer}>
          <View style={styles.avatarContainer}>
            <View style={styles.badge}>
              <Text style={styles.badgeText}>4</Text>
            </View>
            <View>
              <Image
                source={require("assets/images/profile-pic-may.jpg")}
                style={styles.image}
              />
            </View>
          </View>
          <View style={styles.commentBodyContainer}>
            <Text style={styles.commentTitle}>
              Meu pai fez o que ela mandou…
            </Text>
            <Text style={styles.commentDate}>Há 7 minutos atrás</Text>
            <Text>
              Dá certo sim, o meu pai, por exemplo, fugiu quando eu tinha 5 anos
              e eu não faço ideia da onde ele esteja. Nunca mais voltou, pena
              que a violência só piorou.
            </Text>
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  avatarContainer: {
    width: 60
  },
  container: {
    backgroundColor: colours.whiteTransparent,
    borderTopRightRadius: layout.borderRadius,
    padding: layout.spacing,
    width: "100%"
  },
  commentContainer: {
    flex: 1,
    flexDirection: "row",
    marginBottom: layout.spacing
  },
  commentBodyContainer: {
    marginLeft: layout.spacing
  },
  commentDate: {
    color: colours.darkGrey,
    fontStyle: "italic",
    marginBottom: layout.spacing
  },
  commentTitle: {
    fontSize: typography.fontSizeL,
    marginBottom: layout.spacingS
  },
  commentsHeadingContainer: {
    marginBottom: layout.spacing
  },
  commentsHeading: {
    fontSize: typography.fontSizeL,
    fontWeight: "bold"
  },
  badge: {
    alignItems: "center",
    backgroundColor: colours.navyBlueDark,
    borderRadius: 11,
    flexDirection: "column",
    height: 22,
    justifyContent: "center",
    left: 35,
    position: "absolute",
    width: 22,
    zIndex: 1
  },
  badgeText: {
    color: colours.pureWhite,
    fontSize: typography.fontSizeS
  },
  date: {
    fontStyle: "italic"
  },
  titleContainer: {
    backgroundColor: colours.whiteTransparent,
    borderRadius: layout.borderRadius,
    marginBottom: layout.spacing,
    padding: layout.spacing
  },
  title: {
    fontSize: typography.fontSizeL,
    fontWeight: "bold"
  },
  image: {
    backgroundColor: colours.emeraldGreen,
    borderRadius: 25,
    height: 50,
    marginTop: 10,
    overflow: "hidden",
    width: 50
  }
});

export { Comments };
