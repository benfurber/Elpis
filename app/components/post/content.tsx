import React from "react";
import { StyleSheet, Text, View } from "react-native";
import FlexImage from "react-native-flex-image";

import { Post } from "interfaces";
import { colours, elements, layout } from "styles";
import { formatDate, validURL } from "utils";

interface Props {
  content: Post["content"];
  createdAt?: Post["createdAt"];
  imagePath: Post["imagePath"];
}

const Content = (props: Props) => {
  const renderText = () => {
    if (props.content) {
      return <Text style={styles.text}>{props.content}</Text>;
    }
  };

  const renderDate = () => {
    if (props.createdAt) {
      return (
        <Text style={[elements.textDate, styles.text]}>
          {formatDate(props.createdAt)}
        </Text>
      );
    }
  };

  const renderImage = () => {
    const { imagePath } = props;

    if (imagePath && validURL(imagePath)) {
      return <FlexImage source={{ uri: imagePath }} style={styles.image} />;
    }
  };

  return (
    <View style={styles.body}>
      {renderImage()}
      {renderText()}
      {renderDate()}
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: colours.whiteTransparent,
    borderTopRightRadius: layout.borderRadius,
    padding: layout.spacing,
    width: "100%",
  },
  text: {
    marginVertical: 10,
  },
  image: {
    borderRadius: layout.borderRadius,
    overflow: "hidden",
  },
});

export { Content };
