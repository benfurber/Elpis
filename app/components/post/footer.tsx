import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { Avatar } from "components";
import { Author, NavigationType } from "interfaces";
import { colours, layout } from "styles";

interface Props {
  author: Author;
  navigation: NavigationType;
}

const Footer = (props: Props) => {
  const { author, navigation } = props;
  const { avatarPath, id, name } = author;

  return (
    <View>
      <View style={styles.followingContainer}>
        <Text style={styles.text}>{name}</Text>
      </View>
      <View style={styles.avatarContainer}>
        <Avatar
          avatarPath={avatarPath}
          communityId={id}
          containerStyles={styles.authorAvatarContainerStyles}
          navigation={navigation}
          size={"large"}
          styles={styles.authorAvatar}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  authorAvatar: {
    borderColor: colours.navyBlueDark,
    borderWidth: 4,
  },
  authorAvatarContainerStyles: {
    position: "absolute",
    top: -50,
  },
  avatarContainer: {
    alignItems: "flex-end",
    flex: 1,
    flexDirection: "column",
    marginRight: layout.spacing,
  },
  followingContainer: {
    backgroundColor: colours.navyBlueDark,
    borderBottomLeftRadius: layout.borderRadius,
    borderBottomRightRadius: layout.borderRadius,
    flex: 1,
  },
  text: {
    color: colours.pureWhite,
    fontWeight: "bold",
    padding: layout.spacing,
  },
});

export { Footer };
