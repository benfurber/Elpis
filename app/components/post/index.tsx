import React from "react";
import { StyleSheet, Text, View } from "react-native";
import FlexImage from "react-native-flex-image";

import { colours, layout } from "styles";

import { Footer } from "./footer";
import { Tabs } from "./tabs";

interface Props {
  description: string;
}

const Post = (props: Props) => {
  const { description } = props;
  return (
    <View style={styles.container}>
      <Tabs />
      <View style={styles.body}>
        <FlexImage
          source={require("../../assets/images/image_post_1.jpg")}
          style={styles.image}
        />
        <Text style={styles.description}>{description}</Text>
      </View>
      <Footer />
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: colours.whiteTransparent,
    borderTopRightRadius: layout.borderRadius,
    padding: layout.padding,
    width: "100%"
  },
  container: {
    marginBottom: 20
  },
  description: {
    marginVertical: 10
  },
  image: {
    borderRadius: layout.borderRadius,
    overflow: "hidden"
  }
});

export { Post };
