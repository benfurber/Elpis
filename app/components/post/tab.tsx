import React from "react";
import { StyleSheet, TouchableOpacity, View } from "react-native";

import { colours } from "styles";
import { Badge, Icon } from "components";

const iconSize = 30;

interface Props {
  name: string;
  number: null | number;
  onPress: () => void;
  selected: boolean;
  first?: boolean;
}

const Tab = (props: Props) => {
  const { first, name, number, onPress, selected } = props;
  const { whiteTransparent, whiteTransparentHigh } = colours;

  const backgroundColor = selected ? whiteTransparent : whiteTransparentHigh;
  const iconColour = selected ? colours.navyBlueDarkTransparentHigh : null;

  const tabs = {
    flex: 1,
    paddingHorizontal: 16,
    paddingVertical: 8,
  };

  const styles = StyleSheet.create({
    background: {
      ...tabs,
      backgroundColor,
    },
    borderRight: {
      backgroundColor: "transparent",
      borderStyle: "solid",
      borderRightWidth: 10,
      borderBottomWidth: 46,
      borderRightColor: "transparent",
      borderBottomColor: backgroundColor,
    },
    borderLeft: {
      backgroundColor: "transparent",
      borderStyle: "solid",
      borderLeftWidth: 10,
      borderTopWidth: 46,
      borderLeftColor: "transparent",
      borderTopColor: backgroundColor,
    },
    iconCentre: {
      alignSelf: "center",
    },
    shiftContainer: {
      left: first ? 0 : -5,
      flexDirection: "row",
      flex: 1,
    },
  });

  return (
    <View style={styles.shiftContainer}>
      {!first && <View style={styles.borderLeft} />}
      <View style={styles.background}>
        <TouchableOpacity onPress={onPress}>
          <Badge left={52} number={number} />
          <Icon
            colour={iconColour}
            name={name}
            size={iconSize}
            style={styles.iconCentre}
          />
        </TouchableOpacity>
      </View>
      <View style={styles.borderRight} />
    </View>
  );
};

export { Tab };
