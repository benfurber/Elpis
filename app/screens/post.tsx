import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

import { Post } from "components";
import { NavigationType } from "interfaces";
import { colours } from "styles";

interface Props {
  navigation: NavigationType;
}

class PostScreen extends Component<Props> {
  getPost() {
    const { navigation } = this.props;
    return navigation.getParam("post", null);
  }

  getSetDisplay() {
    const { navigation } = this.props;

    let setDisplay;
    if (navigation) {
      setDisplay = navigation.getParam("setDisplay", "body");
    }
    return setDisplay;
  }

  getParam(param: string) {
    const { navigation } = this.props;

    let theParam;
    if (navigation) {
      theParam = navigation.getParam(param);
    }
    return theParam;
  }

  render() {
    const { navigation } = this.props;

    return (
      <View style={styles.container}>
        <Post
          navigation={this.props.navigation}
          post={this.getParam("post")}
          setDisplay={this.getParam("setDisplay")}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colours.transparentBlue,
    flex: 1,
    flexDirection: "column"
  }
});

export { PostScreen };
