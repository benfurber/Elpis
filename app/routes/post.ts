import { PostScreen } from "screens";

const path = "post/:id";
const screen = PostScreen;

const Post = {
  path,
  screen,
};

export { Post };
