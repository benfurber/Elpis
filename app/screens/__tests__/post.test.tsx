import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { UnwrappedPostScreen } from "../post";

let navigation;
jest.mock(navigation, () => jest.fn());

describe("PostScreen", () => {
  describe("when post provided", () => {
    it("renders correctly", () => {
      const post = {
        author: {
          avatarPath: require("assets/images/empower_two_women_logo.png"),
          name: "Empodere Duas Mulheres",
        },
        comments: [],
        createdAt: new Date("2000-01-01"),
        content: null,
        id: "10387-314fs-12asdbj",
        imagePath: require("assets/images/image_post_1.jpg"),
      };

      const component = shallow(
        <UnwrappedPostScreen navigation={navigation} post={post} />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe("when no post provided", () => {
    it("renders correctly", () => {
      const component = shallow(
        <UnwrappedPostScreen navigation={navigation} />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
