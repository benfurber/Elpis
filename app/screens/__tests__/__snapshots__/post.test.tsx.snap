// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostScreen when no post provided renders correctly 1`] = `
<BackgroundContainer>
  <NewQuery
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "Post",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "post",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "directives": Array [],
                        "kind": "FragmentSpread",
                        "name": Object {
                          "kind": "Name",
                          "value": "postAttributes",
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
            ],
          },
          Object {
            "directives": Array [],
            "kind": "FragmentDefinition",
            "name": Object {
              "kind": "Name",
              "value": "postAttributes",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "author",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "directives": Array [],
                        "kind": "FragmentSpread",
                        "name": Object {
                          "kind": "Name",
                          "value": "authorAttributes",
                        },
                      },
                    ],
                  },
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createdAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "content",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "imagePath",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "comments",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "totalReplies",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "typeCondition": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Post",
              },
            },
          },
          Object {
            "directives": Array [],
            "kind": "FragmentDefinition",
            "name": Object {
              "kind": "Name",
              "value": "authorAttributes",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "avatarPath",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
            "typeCondition": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "User",
              },
            },
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 342,
          "start": 0,
        },
      }
    }
    variables={
      Object {
        "id": undefined,
      }
    }
  >
    <Component />
  </NewQuery>
</BackgroundContainer>
`;

exports[`PostScreen when post provided renders correctly 1`] = `
<BackgroundContainer>
  <Post
    post={
      Object {
        "author": Object {
          "avatarPath": Object {
            "testUri": "../../../app/assets/images/empower_two_women_logo.png",
          },
          "name": "Empodere Duas Mulheres",
        },
        "comments": Array [],
        "content": null,
        "createdAt": 2000-01-01T00:00:00.000Z,
        "id": "10387-314fs-12asdbj",
        "imagePath": Object {
          "testUri": "../../../app/assets/images/image_post_1.jpg",
        },
      }
    }
    setDisplay="post"
  />
</BackgroundContainer>
`;
